chain/network-primitives/src/types.rs

```diff
@@ -603,20 +606,20 @@ impl NetworkConfig {
         }
 
         if self.safe_set_size <= self.minimum_outbound_peers {
-            error!(target: "network",
+            bail!(
```

Comment:
can you add a few tests to  test these conditiions where we bail now ?

Reply:
did you add the test?

Reply:
I added a test last month. I rebased the PR.